src/core/border_router/routing_manager.cpp

```diff
@@ -2302,6 +2304,8 @@ void RoutingManager::OmrPrefixManager::DetermineFavoredPrefix(void)
             mFavoredPrefix.SetFrom(prefixConfig);
         }
     }
+
+    Get<MeshCoP::BorderAgent>().PostNotifyMeshCoPServiceChangeTask();
```

Comment:
May be good to add `#if OPENTHREAD_CONFIG_BORDER_AGENT_ENABLE` before these here? 

I expect that `BORDER_AGENT_ENABLE` is likely enabled and used along with `ROUTING_MANAGER` but better to check for it, in case, some products/projects want to implement BA function on their own and not use the OT provided one while using `ROUTING_MANAGER`?

Maybe we can add a method here like `SignalFavoredOmrPrefixChanged()` and then within it we can call `Get<MeshCoP::BorderAgent>().PostNotifyMeshCopServiceChangeTask());` (and check for `BORDER_AGENT_ENABLE` feature). This way we don't need to do `#if` check everywhere and in the future if we need to detect OMR change for any other reason, we have a perfect place for it.

Reply:
Make sense. Added the check.